cmake/Findo3de.cmake

```diff
@@ -21,24 +23,35 @@ o3de_current_file_path(current_path)
 # because later it's read again using a path like ${LY_ROOT_FOLDER}/engine.json, which
 # is also normalized.  They should match to avoid errors on some build systems.
 cmake_path(SET engine_json_path NORMALIZE ${current_path}/../engine.json)
-file(READ ${engine_json_path} engine_json)
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${engine_json_path})
 
-string(JSON this_engine_name ERROR_VARIABLE json_error GET ${engine_json} engine_name)
-if(json_error)
-    message(FATAL_ERROR "Unable to read key 'engine_name' from '${engine_json_path}', error: ${json_error}")
-endif()
+if(NOT LY_ENGINE_NAME_TO_USE)
+    # PACKAGE_VERSION_COMPATIBLE should be set TRUE by o3deConfigVersion.cmake
+    find_package_handle_standard_args(o3de
+        "This engine was not found to be compatible with your project, or no compatibility checks were done. For more information, run the command again with '--log-level VERBOSE'."
+        PACKAGE_VERSION_COMPATIBLE
+    )
+else()
+    # LY_ENGINE_NAME_TO_USE compatibility check for older projects 
+    set(found_matching_engine FALSE)
 
-# Make sure we are matching LY_ENGINE_NAME_TO_USE with the current engine
-set(found_matching_engine FALSE)
-if(this_engine_name STREQUAL LY_ENGINE_NAME_TO_USE)
-    set(found_matching_engine TRUE)
-endif()
+    file(READ ${engine_json_path} engine_json)
+    string(JSON this_engine_name ERROR_VARIABLE json_error GET ${engine_json} engine_name)
+    if(json_error)
+        message(FATAL_ERROR "Unable to read key 'engine_name' from '${engine_json_path}', error: ${json_error}")
+    endif()
 
-find_package_handle_standard_args(o3de
-    "Could not find an engine with matching ${LY_ENGINE_NAME_TO_USE}"
-    found_matching_engine
-)
+    if(this_engine_name STREQUAL LY_ENGINE_NAME_TO_USE)
+        set(found_matching_engine TRUE)
+    else()
+        message(VERBOSE "Project engine name '${LY_ENGINE_NAME_TO_USE}' does not match this engine name '${this_engine_name}' in '${engine_json_path}'")
+    endif()
+
+    find_package_handle_standard_args(o3de
+        "The engine name for this engine '${this_engine_name}' does not match the projects engine name '${O3DE_ENGINE_NAME_TO_USE}'."
```

Comment:
You have `O3DE_ENGINE_NAME_TO_USE` here, so I was a bit confused.

Reply:
oops that's a bug, good catch.